tests/test_core/test_sequence.py

```diff
@@ -2381,6 +2381,27 @@ def test_sliced_seqview_rich_dict(reverse):
     assert rd["offset"] == 2
 
 
+@pytest.mark.parametrize(
+    "sl",
+    (
+        slice(2, 5, 1),  # positive indices, positive step
```

Comment:
would be good to test edge cases, especially for slices at the ends of the sequence.
- if you slice the whole sequence (or the whole seq in reverse), do you get 0 for start, and the length of sequence for end?
- singleton slices, and singleton slices at the start and ends (and reversed).

Reply:
by singleton slice do you mean `slice(i, i+1)`?

Reply:
yep